### PR TITLE
#353: make kokkos required in contact

### LINKED ISSUE #353: make kokkos required in contact code
We have two paths in contact code -- one with Kokkos enabled and one without Kokkos. Kokkos is now a required dependency so we need to clean up that code a bit

### PR DESCRIPTION
**Pull request by nmm0**
Closes #353

### CODE REVIEW

**Comment by nmm0** [comment]
I managed to reproduce the sphere/plate contact error on my local system, but weirdly, it happend when I added ffast-math (pulled in from BVH). I'm not sure what exactly could be causing it here, since we aren't using distBVH in our ci builds, maybe the way in which spack is building something?

**Comment by nmm0** [comment]
I honestly can't get those CI issues to occur again.

**Comment by stmcgovern** [comment]
The initial workflow ran over 30 days ago - so we can't just re-run it again. @nmm0 Do you want to make another commit to have the pipelines run some more to see if the issue re-occurs? Or is this sufficient?

**Comment by nmm0** [comment]
> The initial workflow ran over 30 days ago - so we can't just re-run it again. @nmm0 Do you want to make another commit to have the pipelines run some more to see if the issue re-occurs? Or is this sufficient?

@stmcgovern I ran it with a manual trigger here: https://github.com/NimbleSM/NimbleSM/actions/runs/7543914157 (doesn't show up on the PR)

**Comment by stmcgovern** [comment]
I didn't see that. thanks @nmm0 ! I'll just re-run it again now and then I can review after pipelines finish.

File: src/nimble_cli.cc
Comment (stmcgovern) [inline src/nimble_cli.cc:1]: The cli flag `--use_kokkos` should be removed.

Comment (stmcgovern) [reply src/nimble_cli.cc:1]: def at line 58 and use at line 88 in nimble_cli.cc

File: src/nimble_parser.h
Comment (stmcgovern) [inline src/nimble_parser.h:1]: line 128

File: test/run_exodiff_test.py
Comment (stmcgovern) [inline test/run_exodiff_test.py:1]: I think the cli argument stuff should be removed here:
 ```
   if "use_kokkos" in cli_flag:
        log_file_name = base_name + ".kokkos.np" + str(num_ranks) + ".log"
 ```

Comment (stmcgovern) [reply test/run_exodiff_test.py:1]: Related is the use of the flag in test/CMakeLists.txt and then in the command for the the tests that run (e.g. NimbleSM/test/contact/sphere_plate_contact/CMakeLists.txt at lines 14 and 26)

**Review by stmcgovern — CHANGES_REQUESTED** [review]
I think this looks good. There are some remaining occurrences of the `use_kokkos` flag that I think should be cleaned up for clarity's sake.

**Review by stmcgovern — APPROVED** [review]

**Pull request merged**